Orange/widgets/unsupervised/owmds.py

```diff
@@ -187,135 +181,51 @@ def __init__(self):
+
+        self.graph.jitter_size = self.jitter_size
+        gui.hSlider(box_points, self, "connected_pairs", label="Show similar pairs:", minValue=0,
+                    maxValue=20, createLabel=False, callback=self._on_connected_changed)
+        g.add_widgets(ids=[g.JitterSizeSlider], widget=box_points)
+
+        box_plot_prop = gui.vBox(self.controlArea, "Plot Properties")
+        g.add_widgets([g.ShowLegend,
+                       g.ToolTipShowsAll,
+                       g.ClassDensity,
+                       g.LabelOnlySelected], box_plot_prop)
 
-        box.layout().addLayout(form)
+        self.controlArea.layout().addStretch(100)
+        self.icons = gui.attributeIconDict
+
+        palette = self.graph.plot_widget.palette()
+        self.graph.set_palette(palette)
 
         gui.rubber(self.controlArea)
 
-        box = QGroupBox("Zoom/Select", )
-        box.setLayout(QHBoxLayout())
-        box.layout().setContentsMargins(2, 2, 2, 2)
-
-        group = QActionGroup(self, exclusive=True)
-
-        def icon(name):
-            path = "icons/Dlg_{}.png".format(name)
-            path = pkg_resources.resource_filename(widget.__name__, path)
-            return QIcon(path)
-
-        action_select = QAction(
-            "Select", self, checkable=True, checked=True, icon=icon("arrow"),
-            shortcut=QKeySequence(Qt.ControlModifier + Qt.Key_1))
-        action_zoom = QAction(
-            "Zoom", self, checkable=True, checked=False, icon=icon("zoom"),
-            shortcut=QKeySequence(Qt.ControlModifier + Qt.Key_2))
-        action_pan = QAction(
-            "Pan", self, checkable=True, checked=False, icon=icon("pan_hand"),
-            shortcut=QKeySequence(Qt.ControlModifier + Qt.Key_3))
-
-        action_reset_zoom = QAction(
-            "Zoom to fit", self, icon=icon("zoom_reset"),
-            shortcut=QKeySequence(Qt.ControlModifier + Qt.Key_0))
-        action_reset_zoom.triggered.connect(
-            lambda: self.plot.autoRange(padding=0.1,
-                                        items=[self._scatter_item]))
-        group.addAction(action_select)
-        group.addAction(action_zoom)
-        group.addAction(action_pan)
-        self.addActions(group.actions() + [action_reset_zoom])
-        action_select.setChecked(True)
-
-        def button(action):
-            b = QToolButton()
-            b.setToolButtonStyle(Qt.ToolButtonIconOnly)
-            b.setDefaultAction(action)
-            return b
-
-        box.layout().addWidget(button(action_select))
-        box.layout().addWidget(button(action_zoom))
-        box.layout().addWidget(button(action_pan))
-        box.layout().addSpacing(4)
-        box.layout().addWidget(button(action_reset_zoom))
-        box.layout().addStretch()
-
-        self.controlArea.layout().addWidget(box)
+        self.graph.box_zoom_select(self.controlArea)
 
         box = gui.vBox(self.controlArea, "Output")
         self.output_combo = gui.comboBox(
```

Comment:
We just discussed this with @BlazZupan. There is no universal place for such columns. E.g. in PCA, the components should appear as attributes, and the original attributes as metas. Here, however, x and y can be meta attributes. So: in the intereset of space, remove this combo box and always put coordinates as meta attributes.

Meta attributes are first-class citizens in all visualization widgets. In rare cases, *Select columns* can be used to move them to features.

Reply:
Removed. However, many widgets has an option where one can select an output.